ResearchKit/Common/ORKDirectionView.m

```diff
@@ -0,0 +1,231 @@
+    }
+    
+    CGContextAddPath(context, path);
+    CGContextStrokePath(context);
+    
+    CGContextRestoreGState(context);
+}
+
+@end
+
+
+@interface ORKDirectionView ()
+
+@property (nonatomic, assign) ORKSide orientation;
+@property (nonatomic, strong) ORKArrowView *leftArrow;
+@property (nonatomic, strong) ORKArrowView *middleArrow;
+@property (nonatomic, strong) ORKArrowView *rightArrow;
+@property (nonatomic, assign) NSInteger index;
+@property (nonatomic, strong) NSTimer *timer;
+@property (nonatomic, copy) NSArray *constraints;
+
+@end
+
+
+@implementation ORKDirectionView
+
+- (instancetype)initWithOrientation:(ORKSide)orientation {
+    self = [super initWithFrame:CGRectZero];
+    if (self) {
+        self.opaque = NO;
+        self.orientation = orientation;
+        
+        self.leftArrow = [[ORKArrowView alloc] initWithOrientation:self.orientation];
+        self.leftArrow.translatesAutoresizingMaskIntoConstraints = NO;
+        [self addSubview:self.leftArrow];
+        
+        self.middleArrow = [[ORKArrowView alloc] initWithOrientation:self.orientation];
+        self.middleArrow.translatesAutoresizingMaskIntoConstraints = NO;
+        [self addSubview:self.middleArrow];
+        
+        self.rightArrow = [[ORKArrowView alloc] initWithOrientation:self.orientation];
+        self.rightArrow.translatesAutoresizingMaskIntoConstraints = NO;
+        [self addSubview:self.rightArrow];
+        
+        self.translatesAutoresizingMaskIntoConstraints = NO;
+        [self setNeedsUpdateConstraints];
+    }
+    return self;
+}
+
+- (void)setIndex:(NSInteger)index {
+    _index = index;
+    
+    self.leftArrow.completed = NO;
+    self.middleArrow.completed = NO;
+    self.rightArrow.completed = NO;
+    
+    if (self.orientation == ORKSideLeft) {
+        switch (index) {
+            case 3:
+                self.leftArrow.completed = YES;
+            case 2:
+                self.middleArrow.completed = YES;
+            case 1:
+                self.rightArrow.completed = YES;
+            default:
+                break;
+        }
+    } else {
+        switch (index) {
+            case 3:
+                self.rightArrow.completed = YES;
+            case 2:
+                self.middleArrow.completed = YES;
+            case 1:
+                self.leftArrow.completed = YES;
+            default:
+                break;
+        }
+    }
+}
+
+- (void)didMoveToWindow {
+    if (self.window) {
+        [self startAnimating];
+    } else {
+        [self stopAnimating];
+    }
+}
+
+- (void)startAnimating {
+    [self stopAnimating];
+    self.index = 0;
+    self.timer = [NSTimer scheduledTimerWithTimeInterval:0.5f target:self selector:@selector(incrementIndex) userInfo:nil repeats:YES];
+}
+
+- (void)stopAnimating {
+    [self.timer invalidate];
+    self.timer = nil;
+}
+
+- (void)incrementIndex {
+    self.index = (self.index + 1) % 4;
+}
+
+- (void)updateConstraints {
```

Comment:
The constraints for this view can be made static. I suggest renaming this method to `setUpConstraints` and calling it once, after adding the subviews instead of `setNeedsUpdateConstraints`.